plugin/metrics/prometheus/metricsstore/reader.go

```diff
@@ -18,38 +18,53 @@ import (
 	"context"
 	"time"
 
+	"github.com/prometheus/client_golang/api"
+	"github.com/prometheus/client_golang/api/prometheus/v1"
 	"go.uber.org/zap"
 
 	"github.com/jaegertracing/jaeger/proto-gen/api_v2/metrics"
 	"github.com/jaegertracing/jaeger/storage/metricsstore"
 )
 
 // MetricsReader is a Prometheus metrics reader.
-type MetricsReader struct{}
+type MetricsReader struct {
+	client v1.API
+	logger *zap.Logger
+}
 
 // NewMetricsReader returns a new MetricsReader, assigning the first reachable host:port from the provided list.
 // This host:port forms part of the URL to call when making queries to the underlying metrics store.
-func NewMetricsReader(logger *zap.Logger, hostPorts []string, connTimeout time.Duration) (*MetricsReader, error) {
-	// TODO: Implement me
-	return &MetricsReader{}, nil
+func NewMetricsReader(logger *zap.Logger, hostPort string) (*MetricsReader, error) {
+	client, err := api.NewClient(api.Config{
+		Address: "http://" + hostPort,
+	})
+	if err != nil {
+		return nil, err
+	}
+	mr := &MetricsReader{
+		client: v1.NewAPI(client),
+		logger: logger,
+	}
+	logger.Info("Prometheus reader initialized", zap.String("addr", hostPort))
+	return mr, nil
 }
 
 // GetLatencies gets the latency metrics for the given set of latency query parameters.
-func (m *MetricsReader) GetLatencies(ctx context.Context, params *metricsstore.LatenciesQueryParameters) ([]metrics.Metric, error) {
+func (m *MetricsReader) GetLatencies(ctx context.Context, params *metricsstore.LatenciesQueryParameters) (mf metrics.MetricFamily, err error) {
 	// TODO: Implement me
-	return nil, nil
```

Comment:
nit: we try to stay away from named return values (imo one of the language features where Go really dropped the ball on readability)